kernel/rtlil.cc

```diff
@@ -168,25 +170,24 @@ RTLIL::Const RTLIL::Const::from_string(std::string str)
 std::string RTLIL::Const::decode_string() const
 {
 	std::string string;
-	std::vector<char> string_chars;
-	for (int i = 0; i < int (bits.size()); i += 8) {
+	string.reserve(GetSize(bits)/8);
+	for (int i = 0; i < GetSize(bits); i += 8) {
 		char ch = 0;
 		for (int j = 0; j < 8 && i + j < int (bits.size()); j++)
 			if (bits[i + j] == RTLIL::State::S1)
 				ch |= 1 << j;
 		if (ch != 0)
-			string_chars.push_back(ch);
+			string.append({ch});
```

Comment:
So, doesn't this cause an unnecessary instantiation of an `initializer_list<char>`?  Is it not more efficient to do `string.append(1, ch)`?  Maybe I misunderstand something, because you're even careful enough to use `emplace_back()` instead of `push_back()` above.

Reply:
An `initializer_list<char>` is a pair of pointers to stack; in principle, `append(initializer_list<char>)` should cost the same as `append(int, char)`.